### ISSUE TITLE
what causes my accuracy problems?

### ISSUE THREAD

**Issue by TorHa** [12rajnish/DeePred-BBB#2]
Thanks for a very interesting paper! I had a bit of a struggle getting going with DeePred-BBB and I wonder if any or some of my changes may have caused the prediction to be less accurate. No changes were made in the h5 model file. I can deal with Python code but have no experience in deep learning whatsoever.

- the subprocess.call line required the addition of path+ at several places, like so: subprocess.call(['java', '-jar', path+'/PaDEL-Descriptor/PaDEL-Descriptor.jar','-descriptortypes', path+'/PaDEL-Descriptor/descriptors.xml', '-dir', path, '-file', path+'/feature.csv', '-2d', '-fingerprints', '-removesalt', '-detectaromaticity', '-standardizenitro'])
- an sklearn warning disappeared after adding column titles from the feature file to the data file
- a tensorflow warning disappeared after adding compile=False to the load_model function

After these changes the code is executed and the feature file is updated for every compound. I tested three rounds of 100 randomly chosen compounds from your paper's supplementary data set file. The best accuracy I could get was 0.72. If this is not the expected outcome, any help to get results closer to your reported accuracy would be much appreciated. If my results are perfectly normal I am sure the answer is in your paper but perhaps then at a technical level way above mine. Thanks in advance for any explanation that can help me forward with this!

**Comment by MichaelaBrezinova** [comment]
Hi @TorHa, hope you are doing well! Did you manage to figure this out? I think I might be facing a similar issue. Thanks!

**Comment by TorHa** [comment]
Well, as per above I got the code to execute properly but I've had no feedback on potential accuracy problems. Have had discussions with ADME experts though and they are sceptical to AI efforts to get at the BBB issue. Key problem is that there are way too few compounds where BBB status is unequivocally determined in vivo to form ground truth for models/algorithms. Following the references in the deepred paper (and other similar ones) I can agree that the number is not so impressing but coming from the biology side I can't really say if they are too few for a good deep learning model. Are you also a biologist or do you come from the AI side? Cheers!

**Comment by chushuang203** [comment]
Hello, I defined a DNN model based on the article and found that the training loss is NaN using part of the dataset from the supplementary materials. Would you do me a favor? Thanks!
